sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -149,7 +149,7 @@ case class GetJsonObject(json: Expression, path: Expression)
 
     if (parsed.isDefined) {
       try {
-        Utils.tryWithResource(jsonFactory.createParser(jsonStr.getBytes)) { parser =>
+        Utils.tryWithResource(jsonFactory.createParser(jsonStr.toString)) { parser =>
```

Comment:
I think the root cause is `JacksonParser` -
 https://github.com/apache/spark/blob/a26e3ed5e414d0a350cfe65dd511b154868b9f1d/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala#L370 does not catch the exception (it seems `CharConversionException` is not `RuntimeException`.

I believe this introduces unnecessary string instance.